partition_snapshot_reader.hh

```diff
@@ -148,14 +179,21 @@ class partition_snapshot_flat_reader : public flat_mutation_reader::impl, public
 
         // Valid if has_more_rows()
         const rows_entry& peek_row() const {
-            return *_clustering_rows.front()._position;
+            const auto& r = _clustering_rows.front();
+            return *r._position;
         }
         bool has_more_rows() const {
             return !_clustering_rows.empty();
         }
 
-        const range_tombstone& peek_range_tombstone() const {
-            return _range_tombstones.front().begin()->tombstone();
+        range_tombstone peek_range_tombstone() const {
```

Comment:
Here too.

Reply:
Here the `std::conditional_t` is more sensible and doesn't require implementation changes. If it's fine with you, I would change return type of `peek_` to std::contitional, while leaving the return type of `pop_` as `range_tombstone`.